### ISSUE TITLE
Feature Request: Sort found events desc by date, newest first, when searching for events

### ISSUE THREAD

**Issue by michaelof** [SimpleMobileTools/Simple-Calendar#1882]
SC 6.20.1 pro - F-Droid

Hi Tibor, 

probably small but hopefully not only for me useful request:

I've several appointments/events, which happens pretty often, but not regular like "each ...", so no series.
To add a new appointment, I'm searching for a specific keyword in title/subject, select ALWAYS the latest event, as there is relevant information in "description", needed for new event, and "duplicate" it to the desired time slot.

Currently list of found events is from oldest to newest, so I have to wipe through dozens of events to get the latest. 
Would be great to have this result list sorted desc, showing the latest event at first. 

Regards,
Michael

**Comment by tibbi** [comment]
you mean that you see too many past events? Do you have past event displaying enabled at the list view section of the app settings?

**Comment by michaelof** [comment]
Am 15. November 2022 14:32:59 MEZ schrieb Tibor Kaputa ***@***.***>:
>you mean that you see too many past events? Do you have past event displaying enabled at the list view section of the app settings?
>

I have past event displaying disabled for list view, but (of course) I would expect to see all events matching a given pattern when searching for events via magnifying glass, even if they're in the past.

I'm talking about the search result list of events, they are sorted from oldest to newest, would be cool to have them sorted desc, newest event matching search pattern first in result list.
-- 
Diese Nachricht wurde von meinem Android-Gerät mit K-9 Mail gesendet.

**Comment by tibbi** [comment]
oldest and newest terms doesnt make much sense if you see future events only

**Comment by michaelof** [comment]
> .... if you see future events only

you've confuse me, at first, as I see past events, so tried a little bit more. What I found out is, that there are two different cases:

a) search results exist both in the past and future: Simple-Calendar positions result list to first event in future, as latest entry on page
b) search results exist only for past events. Simple-Calendar positions list the oldest event in result list

--> b) is the case I'm talking about, relevant for me: I frequently need to create future events as (to be modified) duplicate of the latest past event. No other future events existing at this time. When searching e.g. based on pattern for latest event, S-C starts result list with events past in 2016....  would be cool if list would be reverse (date desc) sorted.

Remark for your question about event list view, which I rarely use, I really like your configurable weekly view :)
- Although I have set for list view "Display events from the past: Never", S-C "only" positions list to next future event, but I still can scroll backwards, seeing (greyed out) past events. Not sure if this behaviour is works-as-designed for option "never", but I like it.
Alt

**Comment by e-t-l** [comment]
> a) search results exist both in the past and future: Simple-Calendar positions result list to first event in future, as latest entry on page
> b) search results exist only for past events. Simple-Calendar positions list the oldest event in result list
> 
> --> b) is the case I'm talking about, relevant for me: I frequently need to create future events as (to be modified) duplicate of the latest past event. No other future events existing at this time. When searching e.g. based on pattern for latest event, S-C starts result list with events past in 2016.... would be cool if list would be reverse (date desc) sorted.

I think this is a reasonably common workflow for users who repeat events arbitrarily (i.e. not as a scheduled recurrence). I do the exact same thing, and I agree that it would help users a lot if there were at least an *option* for the Search Results to display past events in newest-to-oldest order. 

Here's an implementation that might be easier to code:

Just like how the normal search results (Michaelof's "a)" scenario, as quoted) displays the full event list past-to-future and positions the view on the next future result, this "b)" scenario could *also* display the full event list past-to-future, but position the view on the most _recent_ result. In both scenarios, the user can scroll upwards to view older and older events.

In other words, whereas the current implementation of the "b)" scenario starts with the view at the top of the search results list (aka oldest event), we would make it start at the bottom of the list (aka most-recent event).


@tibbi We understand that you extremely busy and massively undercompensated for all the amazing work you do here. Would it be reasonable to at least reopen this Issue and give it a https://github.com/SimpleMobileTools/Simple-Calendar/labels/feature%20request label? Just to keep it on the radar. It doesn't have to be a high priority.
